skglm/estimators.py

```diff
@@ -890,7 +899,14 @@ def path(self, X, y, alphas, coef_init=None, return_n_iter=True, **params):
             The number of iterations along the path. If return_n_iter is set to
             ``True``.
         """
-        penalty = compiled_clone(MCPenalty(self.alpha, self.gamma))
+        weights = np.ones(X.shape[1]) if self.weights is None else self.weights
+        if X.shape[1] != len(weights):
+            raise ValueError("The number of weights must match the number of \
+                              features. Got %s, expected %s." % (
+                len(weights), X.shape[1]))
```

Comment:
we better use f-string
```suggestion
            raise ValueError(
                "The number of weights must match the number of features. "
                f"Got {len(weights)}, expected {X.shape[1]}."
            )
```